notebooks/gnn/cgs_creation_extensions/cgs_mag_extension.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# cuGraph or cuGraph-Service is required; each has its own version of
+# import_optional and we need to select the correct one.
+
+from ogb.nodeproppred import NodePropPredDataset
+
+def create_mag(server):
+    dataset = NodePropPredDataset(name = 'ogbn-mag') 
+
+    data = dataset[0]
+
+    # Can't import these before loading MAG; known OGB issue
+    import cudf
+    import dask_cudf
+
+    from cugraph.experimental import MGPropertyGraph
+    from cugraph.experimental import PropertyGraph
+
+    pG = PropertyGraph()
+
+    vertex_offsets = {}
+    last_offset = 0
+
+    for node_type, num_nodes in data[0]['num_nodes_dict'].items():
+        vertex_offsets[node_type] = last_offset
+        last_offset += num_nodes
+        
+        blank_df = cudf.DataFrame({'id':range(vertex_offsets[node_type], vertex_offsets[node_type] + num_nodes)})
+        blank_df.id = blank_df.id.astype('int64')
+        if isinstance(pG, MGPropertyGraph):
+            blank_df = dask_cudf.from_cudf(blank_df, npartitions=2)
```

Comment:
the MAG dataset is different than the _other_ dataset we want to ingest with a similar name (I don't want to say the full name since this is a public repo)

Reply:
2 partitions is sufficient for MAG